kqueen/blueprints/api/views.py

```diff
@@ -179,6 +179,26 @@ def cluster_progress(pk):
     return jsonify(progress)
 
 
+@api.route('/clusters/<uuid:pk>/resize', methods=['PATCH'])
+@jwt_required()
+def cluster_resize(pk):
+    obj = get_object(Cluster, pk, current_identity)
+
+    data = request.json
+    if not isinstance(data, dict) or (isinstance(data, dict) and 'node_count' not in data):
+        abort(400)
+
+    res_status, res_msg = obj.engine.resize(data['node_count'])
```

Comment:
I'd prefer to have `resize` method on cluster instance.

Reply:
We currently do it this way on all views, so maybe we can move it at some later date during refactoring